<?xml version="1.0"?>
<launch>
    <!-- args -->
    <arg name="robot" default="premaidai" />
    <arg name="serial_port" default="/dev/rfcomm0" />
    <arg name="servo_off_mode" default="false" />

    <!-- load robot -->
    <param name="robot_description" command="$(find xacro)/xacro.py '$(find premaidai_description)/robots/$(arg robot).urdf.xacro'" />

    <node name="premaidai_controller_node" pkg="premaidai_controller" type="premaidai_controller_node" output="screen">
        <param name="serial_port" value="$(arg serial_port)" />
        <param name="servo_off_mode" value="$(arg servo_off_mode)" />
        <rosparam command="load" file="$(find premaidai_controller)/config/joint_config.yaml" />
        <remap from="~joint_states" to="/joint_states"/>
    </node>

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />

    <!-- load displays -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find premaidai_description)/launch/display.rviz"/>
</launch>
